### DISCUSSION TITLE
Does caching / automatic deduping apply to prisma requests?

### DISCUSSION THREAD

**Discussion by williamlmao** [vercel/next.js#48304] [Help]
### Summary

From reading the docs, it sounds like the caching / request deduping only works when you use `fetch`, but I have seen some youtube videos that have made requests like this and said they were cached. Maybe they were confused, or maybe I'm misunderstanding the docs. 

In the example below, I make the same request in two server components. Can anyone confirm if this would actually be 1 or 2 requests to the DB? Thanks so much!

```
import { prisma } from "@/db";

export default async function Page({
  params,
}: {
  params: { guildId: string; channelId: string };
}) {
  const users = await prisma.user.findMany();

  return (
    <h1>
      {JSON.stringify(users)}
    </h1>
  );
}
```

```
import { prisma } from "@/db";

export default async function Layout({
  children,
}: {
  children: React.ReactNode;
}) {
  const users = await prisma.user.findMany();
 console.log(users)
  return <div className="">  <h1>
    </h1>{children}</div>;
}
```

**Comment by ciruz** [comment]
I would say its not cached, but maybe Prisma has some caching out of the box, like `react-query` and this is what the youtubers meant by it's cached.

But why not set a timeout for the second request and modify the data inbetween, to verify if the data is cached or not? ⏱️

**Comment by timootten** [comment]
Hey,

did someone tried it out?
Is there a way to dedupe prisma?

Should I call prisma twice or call prisma once and give it with props to the other components?

**Reply by tomdohnal** [reply]
You can wrap your Prisma calls in `cache(...)`
https://nextjs.org/docs/app/building-your-application/data-fetching/caching#per-request-caching

**Reply by pingustar** [reply]
how would you invalidate / revalidate this @tomdohnal?

**Reply by jlucaso1** [reply]
This is at request time, don't need to validate.

**Comment by andrewkouri** [comment]
This seems like a major requirement and gaping hole in the Next.js paradigm since there is no way to pass props (for example a logged in User) from a higher level layout.

Would love to see some workarounds for this!

**Reply by jermahs** [reply]
FWIW yes you can pass the information through. It just has to be serializable. This is what I did
1. extract the user from the session
2. passed a prisma query args object to the client through a prop
3. augmented that prop in the client component (think filters on a table)
4. call the async server function to load the data from the server with the augmented query args

**Comment by CraftsmanTech11** [comment]
I still have this confusion. And even official next js Forum, no one is here to state clearly, that whether it is cached or will not cached. next js team should clearly mention, if only "fetch" api has ability to cache data, and not other database query methods.
